doc/md/language-manual.md

```diff
@@ -2022,6 +2036,17 @@ The context type is a record to allow extension with further fields in future re
 
 :::
 
+Shared functions have different capabilities dependent on their qualification as `shared`, `shared query` or `shared composite query`.
+
+A `shared` function may call any `shared` or `shared query` function, but no `shared composite query` function.
+A `shared query` function may not call any `shared`, `shared query` or `shared composite query` function.
+A `shared composite query` function may call any `shared query` or `shared composite query` function, but no `shared` function.
+
+All varieties of shared functions may call unshared functions.
+
+Composite queries, though composable, can only be called externally (from a frontend) and cannot be initiated from an actor.
+
```

Comment:
Maybe we should mention that each composite query call rolls back its state on function exit, and also does not pass state changes to sub-query or sub-composite-query calls. Therefore, repeated calls (which includes recursive calls) have different semantics than in classical procedural execution (like with update calls).

Reply:
Will do, I was shying away from this ;->

Reply:
Maybe, we could also just reference the general design description of that feature, because it is not a Motoko-specific aspect?

Reply:
I've added some clarification, plus an example.

Reply:
Great explanation!